diff --git a/src/Licensing/Outpost.php b/src/Licensing/Outpost.php
index feeb73ee9..4e59b069c 100644
--- a/src/Licensing/Outpost.php
+++ b/src/Licensing/Outpost.php
@@ -137,6 +137,9 @@ class Outpost
         if ($code == 422) {
             return $this->cacheAndReturnValidationResponse($e);
         } elseif ($code == 429) {
+            if ($this->hasCachedResponse()) {
+                return $this->getCachedResponse();
+            }
             return $this->cacheAndReturnRateLimitResponse($e);
         } elseif ($code >= 500 && $code < 600) {
             return $this->cacheAndReturnErrorResponse();
