diff --git a/resources/js/components/live-preview/LivePreview.vue b/resources/js/components/live-preview/LivePreview.vue
index f4686aad2..6e4cc53ad 100644
--- a/resources/js/components/live-preview/LivePreview.vue
+++ b/resources/js/components/live-preview/LivePreview.vue
@@ -63,7 +63,7 @@
 
                     <transition name="live-preview-contents-slide">
                         <div v-show="panesVisible" ref="contents" class="live-preview-contents items-center justify-center overflow-auto" :class="{ 'pointer-events-none': editorResizing }">
-                            <iframe ref="iframe" frameborder="0" :class="previewDevice ? 'device' : 'responsive'" :style="{ width: previewDeviceWidth, height: previewDeviceHeight }" />
+                            <iframe :src="cp_url('livepreview/loading-screen')" ref="iframe" frameborder="0" :class="previewDevice ? 'device' : 'responsive'" :style="{ width: previewDeviceWidth, height: previewDeviceHeight }" />
                         </div>
                     </transition>
 
diff --git a/resources/views/livepreview/loading-screen.blade.php b/resources/views/livepreview/loading-screen.blade.php
index b9a153eec..3c260a766 100644
--- a/resources/views/livepreview/loading-screen.blade.php
+++ b/resources/views/livepreview/loading-screen.blade.php
@@ -27,7 +27,7 @@
                 </g>
             </svg>
 
-            <div class="mt-1">Aktualisiere Live-Preview</div>
+            <div class="mt-1">Lade Live-Preview</div>
         </div>
     </div> 
 </body>
