diff --git a/resources/js/components/live-preview/LivePreview.vue b/resources/js/components/live-preview/LivePreview.vue
index 3fe026779..43c7167d5 100644
--- a/resources/js/components/live-preview/LivePreview.vue
+++ b/resources/js/components/live-preview/LivePreview.vue
@@ -121,6 +121,7 @@ export default {
             keybinding: null,
             token: null,
             target: 0,
+            loadingUrl: '/services/statamic/cp/livepreview/loading-screen',
         }
     },

@@ -176,16 +177,6 @@ export default {

         canPopOut() {
             return typeof BroadcastChannel === 'function';
-        },
-
-        tokenizedUrl() {
-            let url = this.url;
-
-            url += (url.includes('?') ? '&' : '?') + `target=${this.target}`;
-
-            if (this.token) url += `&token=${this.token}`;
-
-            return url;
         }

     },
@@ -239,8 +230,9 @@ export default {
             source = this.$axios.CancelToken.source();

             this.loading = true;
+            this.updateIframeContents(this.tokenizedUrl(this.loadingUrl));

-            this.$axios.post(this.tokenizedUrl, this.payload, { cancelToken: source.token }).then(response => {
+            this.$axios.post(this.tokenizedUrl(this.url), this.payload, { cancelToken: source.token }).then(response => {
                 this.token = response.data.token;
                 const url = response.data.url;
                 this.poppedOut
@@ -259,6 +251,13 @@ export default {
             if (this.previewDevice) iframe.setAttribute('style', `width: ${this.previewDeviceWidth}; height: ${this.previewDeviceHeight}`);
         },

+        tokenizedUrl(url) {
+            url += (url.includes('?') ? '&' : '?') + `target=${this.target}`;
+            if (this.token) url += `&token=${this.token}`;
+
+            return url;
+        },
+
         close() {
             if (this.poppedOut) this.closePopout();


diff --git a/routes/cp.php b/routes/cp.php
index fcf7a965a..2fa0aeec3 100644
--- a/routes/cp.php
+++ b/routes/cp.php
@@ -26,6 +26,8 @@ Route::middleware('statamic.cp.authenticated')->group(function () {
     Route::get('/', 'StartPageController')->name('index');
     Route::get('dashboard', 'DashboardController@index')->name('dashboard');

+    Route::view('livepreview/loading-screen', 'statamic::livepreview/loading-screen');
+
     Route::get('select-site/{handle}', 'SelectSiteController@select');

     Route::group(['namespace' => 'Structures'], function () {

diff --git a/resources/views/livepreview/loading-screen.blade.php b/resources/views/livepreview/loading-screen.blade.php
index b9a153eec..3c260a766 100644
--- a/resources/views/livepreview/loading-screen.blade.php
+++ b/resources/views/livepreview/loading-screen.blade.php
@@ -27,7 +27,7 @@
                 </g>
             </svg>

-            <div class="mt-1">Aktualisiere Live-Preview</div>
+            <div class="mt-1">Lade Live-Preview</div>
         </div>
     </div>
 </body>
