diff --git a/routes/cp.php b/routes/cp.php
index e1559aef7..75cacd3ac 100644
--- a/routes/cp.php
+++ b/routes/cp.php
@@ -78,6 +78,11 @@ Route::middleware('statamic.cp.authenticated')->group(function () {
                 Route::patch('/', 'EntriesController@update')->name('collections.entries.update');
             });
         });
+
+        // TODO: Should be removed at some point.
+        Route::get('{entry}/{slug}', function ($collection, $entry, $slug) {
+            return redirect($entry->editUrl(), 301);
+        });
     });
 
     Route::group(['namespace' => 'Taxonomies'], function () {
