diff --git a/resources/js/components/fieldtypes/bard/BardFieldtype.vue b/resources/js/components/fieldtypes/bard/BardFieldtype.vue
index 2805e3faf..a181fd933 100644
--- a/resources/js/components/fieldtypes/bard/BardFieldtype.vue
+++ b/resources/js/components/fieldtypes/bard/BardFieldtype.vue
@@ -289,8 +289,9 @@ export default {
             extensions: this.getExtensions(),
             content: content,
             editable: !this.readOnly,
-            disableInputRules: ! this.config.enable_input_rules,
-            disablePasteRules: ! this.config.enable_paste_rules,
+            // Polygon TLP-1574: Remove only the input and paste rules for italics
+            disableInputRules: ['italic'],
+            disablePasteRules: ['italic'],
             onInit: ({ state }) => {
                 if (content !== null && typeof content === 'object') {
                     try {
