diff --git a/src/Facades/Endpoint/URL.php b/src/Facades/Endpoint/URL.php
index 7cfbcb805..40cd80dd2 100644
--- a/src/Facades/Endpoint/URL.php
+++ b/src/Facades/Endpoint/URL.php
@@ -242,7 +242,8 @@ class URL
             return config('app.url');
         }
 
-        $rootUrl = app('request')->root();
+        // https://github.com/statamic/cms/pull/8659
+        $rootUrl = url()->to('/');
 
         return Str::ensureRight($rootUrl, '/');
     }
