diff --git a/resources/js/plugins/can.js b/resources/js/plugins/can.js
index d82b33d25..82a10c93c 100644
--- a/resources/js/plugins/can.js
+++ b/resources/js/plugins/can.js
@@ -1,5 +1,5 @@
 const can = function(permission) {
-    const permissions = JSON.parse(atob(Statamic.$config.get('permissions')));
+    const permissions = Statamic.$config.get('permissions') ? JSON.parse(atob(Statamic.$config.get('permissions'))) : [];
 
     return permissions.includes('super') || permissions.includes(permission);
 };
