diff --git a/resources/lang/de/validation.php b/resources/lang/de/validation.php
index 71bd4fc87..63b0187b9 100644
--- a/resources/lang/de/validation.php
+++ b/resources/lang/de/validation.php
@@ -89,6 +89,8 @@ return [
     'unique_user_value' => 'Dieser Wert wurde bereits vergeben.',
     /* Polygon TLP-1683: TODO - Check if we can solve the part from CMS */
     'unique_value' => 'Dieser Wert wurde bereits vergeben.',
+    /* Polygon TLP-2443: TODO - Check if we can solve the part from CMS */
+    'uuid_or_deleted' => 'Dieser Wert muss eine gültige UUID oder "delete" sein.',
     'duplicate_field_handle' => 'Feld mit dem Handle :handle kann nicht mehr als einmal verwendet werden.',
     'one_site_without_origin' => 'Mindestens eine Website darf keine Quelle enthalten.',
     'origin_cannot_be_disabled' => 'Kann keine deaktivierte Quelle auswählen.',
