diff --git a/resources/js/components/Echo.js b/resources/js/components/Echo.js
index 3a2761b6f..8e0b7a12b 100644
--- a/resources/js/components/Echo.js
+++ b/resources/js/components/Echo.js
@@ -22,6 +22,7 @@ class Echo {
         };
 
         this.echo = new LaravelEcho(config);
+        window.addEventListener('unload', () => this.echo.disconnect());
 
         this.bootedCallbacks.forEach(callback => callback(this));
         this.bootedCallbacks = [];
