diff --git a/resources/js/components/fieldtypes/bard/Set.vue b/resources/js/components/fieldtypes/bard/Set.vue
index d295c71a8..054786dcb 100644
--- a/resources/js/components/fieldtypes/bard/Set.vue
+++ b/resources/js/components/fieldtypes/bard/Set.vue
@@ -161,6 +161,9 @@ export default {
         },
 
         previewUpdated(handle, value) {
+            // Polygon TLP-2579: Collaboration - when adding a new set, this.preview is undefined for all other users
+            if (!this.previews) return;
+
             let previews = clone(this.previews);
             previews[handle] = value;
             this.options.bard.updateSetPreviews(this.node.attrs.id, previews);
diff --git a/resources/js/components/fieldtypes/replicator/Set.vue b/resources/js/components/fieldtypes/replicator/Set.vue
index 365567fa4..e037e99c9 100644
--- a/resources/js/components/fieldtypes/replicator/Set.vue
+++ b/resources/js/components/fieldtypes/replicator/Set.vue
@@ -38,7 +38,8 @@
             </div>
         </div>
 
-        <div class="replicator-set-body" v-if="!collapsed">
+        <!-- Polygon TLP-2579: Collaboration - when adding a new set meta is undefined for all other users -->
+        <div class="replicator-set-body" v-if="!collapsed && meta">
             <set-field
                 v-for="field in fields"
                 v-show="showField(field, fieldPath(field))"
